js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetsGroupedViewV2.tsx

```diff
@@ -0,0 +1,237 @@
+import {Box, Colors, Container, Icon, IconName, Inner, Tab, Tabs} from '@dagster-io/ui-components';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import React, {useMemo, useRef} from 'react';
+import {useCatalogViews} from 'shared/assets/catalog/useCatalogViews.oss';
+
+import {
+  AssetSelectionSummaryListItem,
+  AssetSelectionSummaryListItemFromSelection,
+} from './AssetSelectionSummaryListItemV2';
+import {ViewType, getGroupedAssets} from './util';
+import {COMMON_COLLATOR} from '../../app/Util';
+import {KNOWN_TAGS, KnownTagType} from '../../graph/OpTags';
+import {useQueryPersistedState} from '../../hooks/useQueryPersistedState';
+import {InsightsIcon} from '../../insights/InsightsIcon';
+import {LoadingSpinner} from '../../ui/Loading';
+import {weakMapMemoize} from '../../util/weakMapMemoize';
+import {AssetTableFragment} from '../types/AssetTableFragment.types';
+import {useAllAssets} from '../useAllAssets';
+
+const PROPERTIES = ['owners', 'groupName', 'repository', 'tags', 'kinds'] as const;
+type Property = (typeof PROPERTIES)[number];
+
+export const AssetsGroupedView = ({assets}: {assets: AssetTableFragment[]}) => {
+  const {privateViews, publicViews, loading: loadingCatalogViews} = useCatalogViews();
+
+  const groupedAssets = useMemo(() => getGroupedAssets(assets), [assets]);
+
+  const [selectedTab, setSelectedTab] = useQueryPersistedState<Property | 'selections'>({
+    queryKey: 'tab',
+    behavior: 'push',
+    encode: (value) => ({tab: value}),
+    decode: (qs) => {
+      if (typeof qs.tab === 'string' && PROPERTIES.includes(qs.tab as Property)) {
+        return qs.tab as Property;
+      }
+      return 'selections';
+    },
+  });
+
+  const tabs = useMemo(() => {
+    const ossTabs = Object.entries(propertyToLabelAndIcon).map(([key]) => {
+      const {label} = propertyToLabelAndIcon[key as Property];
+      return <Tab key={key} id={key} title={label} />;
+    });
```

Comment:
Using `Object.entries`, you have both the key and the value available, which means you don't need to reach back into `propertyToLabelAndIcon` at all.

```
const ossTabs = Object.entries(propertyToLabelAndIcon).map(([key, entry]) => {
  const {label} = entry;
  return <Tab key={key} ... />;
};